libr/asm/arch/arm/armass.c

```diff
@@ -330,6 +330,60 @@ static int getlist(char *op) {
 	return list;
 }
 
+static ut32 thumb_getshift(const char *str) {
+	// only immediate shifts are ever used by thumb-2. Bit positions are different from ARM.
+	const char *shifts[] = {
+		"LSL", "LSR", "ASR", "ROR", 0, "RRX"
+	};
+	char *type;
+	char *arg;
+	char *space;
+	ut32 res = 0;
+	ut32 shift = false;
+	err = false;
+	ut32 argn;
+	ut32 i;
+	
+	type = strdup(str);
+
+	r_str_case(type,true);
+	
+	if (strcmp(type, shifts[5])) {
```

Comment:
dont use negated logic, it makes code harder to read. use if (!strcmp (

also add a space before every (

Reply:
it's just a bug - fixed in next commit